upgrade_versions.py

```diff
@@ -239,6 +239,8 @@ def push_new_versions(versions_repo, release_date, versions_repo_push_url,
     refspec = "HEAD:refs/heads/{}".format(versions_repo_push_branch)
     push_info = remote.push(refspec=refspec)[0]
     LOG.debug("Push result: {}".format(push_info.summary))
+    if git.PushInfo.ERROR & push_info.flags:
+        raise repository.PushError(push_info)
 
```

Comment:
This is only separated because I've found about it later and so I just created another commit. Squashing.